File: Examples/R/tests/testPlanning.R
test_that("CLP planning solves the TWD_land LP scenario", {
  skipIfFMTFeatureMissing("OSI")
  skipIfSolverInterfaceMissing()

  solverInterface <- fmtSolverInterface()

  if (is.null(solverInterface$CLP)) {
    skip("CLP is not available in the exported solver interface")
  }

  primary <- fmtModelPath()
  parser <- new(FMTmodelparser)

  models <- parser$readproject(
    primary,
    c("LP"),
    TRUE,
    TRUE,
    TRUE
  )

  expect_length(models, 1L)

  model <- new(
    FMTlpmodel,
    models[[1]],
    solverInterface$CLP
  )

  model$setintparameter(
    FMTintmodelparameters$LENGTH,
    2L
  )

  solved <- model$doplanning(TRUE, list())
  expect_true(solved)

  objective <- model$getobjectivevalue()
  expect_true(is.finite(objective))

  outputs <- model$getoutputs()
  expect_gt(length(outputs), 0L)
})

File: Examples/Python/tests/testPlanning.py
import math
import pytest
from FMT import Models, Parser
from conftest import requireFeature


@pytest.mark.requiresOsi
def testClpPlanningWorkflow(modelPath):
    requireFeature("OSI")
    parser = Parser.FMTmodelparser()
    parser.setquietlogger()
    models = parser.readproject(str(modelPath), ["LP"])
    assert len(models) == 1

    model = Models.FMTlpmodel(models[0], Models.FMTsolverinterface.CLP)
    assert model.setparameter(Models.FMTintmodelparameters.LENGTH, 2)
    solved = model.doplanning(True)
    assert solved

    objective = model.getobjectivevalue()
    assert math.isfinite(objective)
    outputs = model.getoutputs()
    assert len(outputs) > 0
